src/cleanfigure.m

```diff
@@ -237,39 +237,31 @@ function pruneOutsideBox(meta, handle)
       % 'loose' ends may now appear in the figure.
       % To avoid this, add a row of NaNs wherever a block of actual data is
       % removed.
-      chunkIndices = [];
-      k = 1;
-      while k <= numPoints
-          % fast forward to shouldPlot==True
-          while k<=numPoints && ~shouldPlot(k)
-              k = k+1;
-          end
-          kStart = k;
-          % fast forward to shouldPlot==False
-          while k<=numPoints && shouldPlot(k)
-              k = k+1;
-          end
-          kEnd = k-1;
 
-          if kStart <= kEnd
-              chunkIndices = [chunkIndices; ...
-                              [kStart, kEnd]];
-          end
+      % Get the indices of points that should not be plotted
+      id_remove = find(shouldPlot==0);
```

Comment:
Line 220 defines,  `shouldPlot = false(numPoints, 1);` and subsequent calls are logical operations, if there are NaNs, we should actually investigate why